server/routes/api/messages.js

```diff
@@ -13,9 +13,37 @@ router.post("/", async (req, res, next) => {
 
     // if we already know conversation id, we can save time and just add it to message and return
     if (conversationId) {
-      const message = await Message.create({ senderId, text, conversationId });
-      return res.json({ message, sender });
+      // check that conversationId belongs to the senderId + recipientId
+
+      // get conversation by id
+      let conversation = await Conversation.findConversationById(
+        conversationId
+      );
+
+      // conversation id is invalid
+      if (!conversation) {
+        return res.sendStatus(404);
+      }
+
+      // sender and recipient are indeed part of the conversation
+      if (
+        (conversation.user1Id == senderId &&
+          conversation.user2Id == recipientId) ||
+        (conversation.user1Id == recipientId &&
+          conversation.user2Id == senderId)
+      ) {
+        const message = await Message.create({
+          senderId,
+          text,
+          conversationId,
+        });
+        return res.json({ message, sender });
+      }
+
+      // conversation id is not valid for given senderId and recipientId
+      return res.sendStatus(400);
```

Comment:
Might be better to send back a 403 instead. [Explaination](https://stackoverflow.com/questions/29912305/response-code-400-or-403-for-post-restful-apis)